src/subword.rs

```diff
@@ -107,6 +108,29 @@ impl SubwordIndices for str {
 
         indices
     }
+
+    fn ngrams_indices(&self, min_n: usize, max_n: usize, buckets_exp: usize) -> Vec<(String, u64)> {
```

Comment:
This shares most code with `subword_indices`. The overlap should be factored out. Maybe it would be possible to create an iterator that produces `(&str, u64)` iterms. `ngram_indices` could just collect these, `subword_indices` could `map` over the iterator, only retaining the indices. With inlining/optimizations, the actual string slices would probably never be produced by `subword_indices`.